florist/api/models/mnist.py

```diff
@@ -29,3 +33,12 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         x = x.view(-1, 16 * 4 * 4)
         x = f.relu(self.fc1(x))
         return f.relu(self.fc2(x))
+
+    @classmethod
+    def mandatory_server_info_fields(cls) -> List[str]:
```

Comment:
This seems to be beyond the scope of just model specific information and more to do with general experiment details. But this class is also tightly coupled with being a model class since it extends nn.Module. I don't know if we can route this info through elsewhere or expand the scope of the class to not be so model centric. Let me know what you think!

Reply:
That's a good point. I was not 100% on board with the solution while I was doing it.

The thing is, the parameters sent to the server through `server_info` are dependent on what "strategy" we are using (like APFL, SCAFFOLD, FedPer, etc. they all have different parameters in their config) but they all use the same `FLServer` and different clients. Sometimes they have different models, like FedPer for example, so I felt this was what I could do to make it the least out-of-scope possible.

I wanted a way to validate the config before starting the training itself but I don't know what is the best way of doing it. Making the client validate those seems a bit weird since those parameters have more to do with the server. Maybe I should just let training fail and display the error, let the user decide what is missing, but that feels like it's not the best way of doing it either.

What do you think?

Reply:
Can we have a server info validator that takes in server info and validates that the required keys for all experiments like server rounds. batch size and local epochs. Then once we start supporting more strategies, we can have the server info validate take in the strategy type and validate the additional parameters specific to the strategy are present

Reply:
I see we already have a method in the job class, maybe that can be a good place

Reply:
The job might not be the best place to put this, but I like the idea of a validator.

Do you think we should add a new attribute to the job that will tell us which validator to use for that training job? Maybe make it optional and the default could be a "basic" validator that will check for `n_server_rounds`, `batch_size` and `local_epochs`.

Reply:
Yeah that sounds good to me!

Reply:
Done the refactor. I called it a parser instead because it's doing the validation and the parsing all at the same time. Also, I used the opportunity to rename`server_info` into `server_config` because I think it's a better name. Please check again.

Reply:
I like that you packed those two together and did the renaming as well, I think it makes sense!